### PR TITLE
[DO NOT MERGE] FFT - _transactions2, Part ??: Proof of Concept for Fresh Stacks

### PR DESCRIPTION
**Pull request by jeremyk-91**
This is a proof of concept for _transactions2 as installed on a fresh stack. It does not include migration at all, but can be used for benchmarking and discussion.

On this branch, we use _transactions2 exclusively for handling transactions. We observe that AtlasDB's integration and end-to-end tests are able to pass.

This is not suitable for merging, obviously, as it will wreck stacks with any existing data.

This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3393)

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/cas/CheckAndSetQueries.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/cas/CheckAndSetQueries.java]: - [ ] License

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/cas/CheckAndSetResponseDecoder.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/cas/CheckAndSetResponseDecoder.java]: - [ ] License

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/cas/CheckAndSetResult.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/cas/CheckAndSetResult.java]: - [ ] License

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/cas/CheckAndSetRunner.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/cas/CheckAndSetRunner.java]: - [ ] License

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/AbstractKeyValueServiceBackedTransactionService.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/AbstractKeyValueServiceBackedTransactionService.java]: - [ ] License

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/DynamicSplittingTransactionService.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/DynamicSplittingTransactionService.java]: - [ ] License

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/v2/TicketingTransactionService.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/v2/TicketingTransactionService.java]: - [ ] License

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/service/DynamicSplittingTransactionServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/service/DynamicSplittingTransactionServiceTest.java]: - [ ] License

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/service/v2/TicketingTransactionServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
```
Comment (felixdesouza) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/service/v2/TicketingTransactionServiceTest.java]: - [ ] License

**Comment by gmaretic** [comment]
Subsumed by all the _transactions2 PRs

**Pull request closed without merging**